src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs

```diff
@@ -275,7 +304,7 @@ impl HummockLevelsExt for Levels {
                 deleted = level_delete_ssts(&mut self.levels[idx], &delete_sst_ids_set) || deleted;
             }
         }
-        if local_related_only && !deleted {
+        if local_related_only && !delete_sst_ids_set.is_empty() && !deleted {
```

Comment:
Could you explain why !delete_sst_ids_set.is_empty()?

Reply:
> Could you explain why !delete_sst_ids_set.is_empty()?

logic of codes here is that: 
- (a) This is a compaction
- (b) generated SSTs are from `delete_sst_ids_set`
so if we pay no attention to SSTs in `delete_sst_ids_set`, we shall also pay no attention to generated SSTs

However, if `delete_sst_ids_set` is empty, Condition (b) is not maintained

Reply:
> Could you explain why !delete_sst_ids_set.is_empty()?

> Could you explain why !delete_sst_ids_set.is_empty()?

logic of codes here is that: 
- (a) This is a compaction
- (b) generated SSTs are from `delete_sst_ids_set`

Therefore if we pay no attention to SSTs in `delete_sst_ids_set`, we shall also pay no attention to generated SSTs

However, if `delete_sst_ids_set` is empty, Condition (b) is not maintained